deepchem/models/tests/test_layers.py

```diff
@@ -1204,3 +1204,37 @@ def test_edge_network():
         np.array(torch_result),
         np.load("deepchem/models/tests/assets/edgenetwork_result.npy"),
         atol=1e-04)
+
+
+@pytest.mark.torch
+def test_mxmnet_global_message_passing():
+    """ Test for MXMNetGlobalMessagePassing Layer."""
+    seed = 123
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+
+    dim = 1
+    node_features = torch.tensor([[0.8343], [1.2713], [1.2713], [1.2713],
+                                  [1.2713]])
+
+    edge_attr = torch.tensor([[1.0004], [1.0004], [1.0005], [1.0004], [1.0004],
+                              [-0.2644], [-0.2644], [-0.2644], [1.0004],
+                              [-0.2644], [-0.2644], [-0.2644], [1.0005],
+                              [-0.2644], [-0.2644], [-0.2644], [1.0004],
+                              [-0.2644], [-0.2644], [-0.2644]])
+
+    edge_indices = torch.tensor(
+        [[0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4],
+         [1, 2, 3, 4, 0, 2, 3, 4, 0, 1, 3, 4, 0, 1, 2, 4, 0, 1, 2, 3]])
+
+    out = dc.models.torch_models.layers.MXMNetGlobalMessagePassing(
+        dim, activation_fn='silu')
+    output = out(node_features, edge_attr, edge_indices)
+    output = output.detach().numpy()
+    result = np.array([[-0.27947044], [2.417905], [2.417905], [2.4178727],
```

Comment:
How did you compute this result? Was it done by hand?

Reply:
After checking that the architecture of the layer implemented and the one in the original code base are same. I just asserted the result I got.
I tried using manual_seed to exactly compare both the layers side by side. But though the implementation of MLP in original code base is through list comprehension and is different from one in deepchem, the weights are initialized differently. I wanted to discuss about it in the meeting. Can you suggest something for this?